python/pyspark/sql/connect/session.py

```diff
@@ -418,6 +426,28 @@ def createDataFrame(
             # If no schema supplied by user then get the names of columns only
             if schema is None:
                 _cols = [str(x) if not isinstance(x, str) else x for x in data.columns]
+                infer_pandas_dict_as_map = (
+                    str(self.conf.get("spark.sql.execution.pandas.inferPandasDictAsMap")).lower()
+                    == "true"
+                )
+                if infer_pandas_dict_as_map:
```

Comment:
will this code path be reused in vanilla pyspark (arrow optimization enabled)?

Reply:
Oh, I think we should make it available for vanilla PySpark as well. Let me add some more UTs for vanilla PySpark to make sure.

Reply:
Updated!